src/mono/mono/metadata/metadata-internals.h

```diff
@@ -943,6 +976,18 @@ mono_metadata_clean_generic_classes_for_image (MonoImage *image);
 MONO_API void
 mono_metadata_cleanup (void);
 
+#ifndef ENABLE_METADATA_UPDATE
+static inline gboolean
+mono_metadata_table_bounds_check (MonoImage *image, int table_index, int token_index)
+{
+	/* token_index is 1-based. TRUE means the token is out of bounds */
+	return token_index > image->tables [table_index].rows;
+}
+#else
+gboolean
+mono_metadata_table_bounds_check (MonoImage *image, int table_index, int token_index);
+#endif
```

Comment:
This and others might need to be inline since they are perf critical.

Reply:
Ok inlined the fast path for `mono_metadata_table_bounds_check` and for `mono_image_effective_table`